### PR TITLE
added segment control inside navigationItem.titleView

### PR DESCRIPTION
**Pull request by jayhu93**

### CODE REVIEW

File: StarWarsAPI-Code/NamesTableViewController.swift [hunk 1]
```diff
@@ -3,15 +3,27 @@ import UIKit
 class NamesTableViewController: UITableViewController {
 
     var names = ["Douglas", "Jay", "Gloria", "John"]
+    var tableSegments = ["Names", "Spaceships"]
     let nameCell = "NameCell"
 
+
     override func viewDidLoad() {
-        title = "Star Wars Names"
+//        title = "Star Wars Names"
```
Comment (regnerjr) [inline StarWarsAPI-Code/NamesTableViewController.swift]: remove commented line.

File: StarWarsAPI-Code/NamesTableViewController.swift [hunk 2]
```diff
@@ -3,15 +3,27 @@ import UIKit
...
+        
         tableView.register(UITableViewCell.self, forCellReuseIdentifier: nameCell)
         Network.getNames { [weak self] (names) in
             self?.names = names
             self?.tableView.reloadData()
         }
+        let segmentControl = UISegmentedControl(items: tableSegments)
+        segmentControl.selectedSegmentIndex = 0
+        navigationController?.navigationBar.backgroundColor = UIColor.blue
```
Comment (regnerjr) [inline StarWarsAPI-Code/NamesTableViewController.swift]: remove please 😄

File: StarWarsAPI-Code/NamesTableViewController.swift [hunk 3]
```diff
@@ -3,15 +3,27 @@ import UIKit
...
     override func viewDidLoad() {
-        title = "Star Wars Names"
+//        title = "Star Wars Names"
+        
         tableView.register(UITableViewCell.self, forCellReuseIdentifier: nameCell)
         Network.getNames { [weak self] (names) in
             self?.names = names
             self?.tableView.reloadData()
         }
+        let segmentControl = UISegmentedControl(items: tableSegments)
```
Comment (regnerjr) [inline StarWarsAPI-Code/NamesTableViewController.swift]: Might be cool to move all this segmented control config into a lazy closure?

**Comment by jayhu93** [comment]
fixed! please review @regnerjr 👼

File: StarWarsAPI-Code/NamesTableViewController.swift [hunk 4]
```diff
@@ -7,14 +7,23 @@ extension UITableViewCell: CellConfigurable {
...
+    override func viewDidLoad() {       
         super.viewDidLoad()
         title = "Star Wars Names"
         Network.getNames { [weak self] (names) in
-            self?.data = names
-            self?.tableView.reloadData()
+        self?.data = names
+        self?.tableView.reloadData()
         }
+        segmentControl.addTarget(self, action: #selector(NamesTableViewController.segmentDidChange(sender:)), for: .valueChanged)
```
Comment (regnerjr) [inline StarWarsAPI-Code/NamesTableViewController.swift]: @yhu98 Could you add this to the to the initializer as well?

Comment (jayhu93) [reply StarWarsAPI-Code/NamesTableViewController.swift]: sure thing!

**Pull request merged**